### PR TITLE
Adds Aerospike Integration #3273

### LINKED ISSUE #3273: [New Integration]: Integration for open-source No-SQL Database : Aerospike
### Is there an existing integration?

- [X] I have searched the existing integrations.

### Use Case

logo-launchpad-400x198

Aerospike is an open source distributed database. Aerospike is built on a "shared nothing" architecture designed to reliably store terabytes of data with automatic fail-over, replication and cross data-center synchronization.

Aerospike overcomes vendor lock-in and is easily deployed in any public, private or multi-cloud environment, is available as a managed service and is plug-compatiblewith your existing environment.

### Motivation

- The multi-model, NoSQL Aerospike Database 6 enables organizations to cost-effectively process gigabytes-to-petabytes of data to make real-time decisions with unlimited scale

- Unlike traditional legacy NoSQL ystems, Aerospike delivers predictable performance, scales from gigabytes to petabytes, is strongly consistent, with unparalleled cross-datacenter replication for a true globally distributed real-time database

- Unlimited scale

- Reduce footprint, grow business
Screenshot 2022-10-10 at 6 40 38 PM

### Implementation

Aerospike is currently expanding its support for several new databases in the market. The most common way to set a new database integration for Aerospike is to open a new issue on their Github. Then Aerospike and MindsDB teams can come together to finalize an approach to implement the integration.

Support Documentation for its implementation using Python Libraries can be found here at :https://developer.aerospike.com/client/python

### Anything else?

Github Aerospike  : https://github.com/aerospike
Use case Aerospike with AI/ML : https://aerospike.com/solutions/use-cases/ai-ml/

Thanks for going through the issue. If you like the idea, then kindly upvote by reacting with a 👍 emoji below.

### PR DESCRIPTION
**Pull request by bispaul**
## Description

Please include a summary of the change and the issue it solves. 

Fixes #3273 [New Integration]: Integration for open-source No-SQL Database : Aerospike

## Type of change

(Please delete options that are not relevant)

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📢 Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] 📄 This change requires a documentation update

## Verification Process

To ensure the changes are working as expected:

 - [ ]   Test Location: Specify the URL or path for testing.
 - [x]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

## Additional Media:

- [ ] I have attached a brief loom video or screenshots showcasing the new functionality or change.

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have appropriately commented on my code, especially in complex areas.
- [x] Necessary documentation updates are either made or tracked in issues.
- [ ] Relevant unit and integration tests are updated or added.

## Testing Aerospike Handler with home_rentals data  

**1: Load the home_rentals_new.csv to Aerospike
[home_rentals_new.csv](https://github.com/mindsdb/mindsdb/files/13064517/home_rentals_new.csv)
[home_rental.json](https://github.com/mindsdb/mindsdb/files/13064520/home_rental.json)

Refer https://docs.aerospike.com/tools/asloader

```./run_loader -h 127.0.0.1 -p 3000 -c home_rental.json ../home_rentals_new.csv```

Screenshot 2023-10-23 at 2 01 17 AM

**2: Login to Aerospike DB and confirm

Screenshot 2023-10-23 at 2 05 07 AM

**3. . Testing Kinetica CREATE DATABASE**
```
CREATE DATABASE aerospike_db
WITH ENGINE = "aerospike",
PARAMETERS = {
    "user": "test",
    "password": "password",
    "host": "localhost",
    "port": 3000,
    "namespace": "test"
    };
```

Screenshot 2023-10-22 at 8 42 43 PM

Screenshot 2023-10-22 at 8 43 13 PM

**5. Testing CREATE PREDICTOR**
```
CREATE MODEL 
  mindsdb.aerospike_home_rentals_model
FROM aerospike_db
  (SELECT * FROM  house_rentals)
PREDICT rental_price;
```

Screenshot 2023-10-22 at 8 44 43 PM

**5. Testing FIND STATUS OF PREDICTOR**
```
DESCRIBE mindsdb.aerospike_home_rentals_model;

SHOW MODELS
FROM mindsdb
LIKE 'aerospike_home_rentals_model'
WHERE status = 'complete';
```

Screenshot 2023-10-22 at 8 45 11 PM

Screenshot 2023-10-22 at 8 46 22 PM

**6. Testing PREDICTOR**
```
SELECT rental_price, 
       rental_price_explain 
FROM mindsdb.aerospike_home_rentals_model
WHERE sqft = 823
AND location='good'
AND neighborhood='downtown'
AND days_on_market=10;

SELECT t.rental_price as real_price, 
m.rental_price as predicted_price,
t.numberofrooms,  t.numberofbthrms, t.sqft, t.location
FROM aerospike_db.house_rentals as t 
JOIN mindsdb.aerospike_home_rentals_model as m
LIMIT 100;
```

Screenshot 2023-10-22 at 8 46 39 PM

Screenshot 2023-10-22 at 8 47 08 PM

**7: Unit Tests**

```
cd mindsdb/integrations/handlers/aerospike_handler/tests
python test_aerospike_handler.py
```

Screenshot 2023-10-23 at 6 41 18 PM

Drop a remark based on your observation.

-   [x] Works Great 💚 (This means that all the steps were executed successfuly and the expected outputs were returned.)
-   [ ] There's a Bug 🪲 [Issue Title](URL To the Issue you created) ( This means you encountered a Bug. Please open an issue with all the relevant details with the Bug Issue Template)

---

### CODE REVIEW

**Comment by bispaul** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by martyna-mindsdb** [comment]
@bispaul 
Please implement unit tests for this handler.

**Comment by bispaul** [comment]
Added the unit tests for this handler.

File: mindsdb/integrations/handlers/aerospike_handler/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+-   `password` is the database password.
+-   `host` is the host IP address or URL.
+-   `port` is the port used to make TCP/IP connection.
+-   `namespace` is the aerospike namespace.
+
+Other optional parameters are not supported as of now.
+
+## Usage
+
+In order to make use of this handler and connect to the MariaDB database in MindsDB, the following syntax can be used:
```
[the remark this answers is not in the document]
Comment (bispaul) [reply mindsdb/integrations/handlers/aerospike_handler/README.md]: Done. Replaced MariaDB with Aerospike.

**Review by martyna-mindsdb — APPROVED** [review]

**Comment by bispaul** [comment]
Hi @martyna-mindsdb , those flake8 errors not from the Aerospike handler.

Screenshot 2023-10-23 at 11 46 32 PM

**Comment by martyna-mindsdb** [comment]
@bispaul 
These autogluon errors are probably on staging. Please try to pull from staging.

**Comment by bispaul** [comment]
Hi @martyna-mindsdb have done a pull from staging

**Comment by bispaul** [comment]
Hi @martyna-mindsdb have done another pull from staging. There is a issue of testing with openai.. I have no idea about that.

**Pull request merged**